sdk/python/feast/infra/offline_stores/bigquery.py

```diff
@@ -488,10 +488,24 @@ def to_bigquery(
             return str(job_config.destination)
 
         with self._query_generator() as query:
-            self._execute_query(query, job_config, timeout)
+            dest = job_config.destination
```

Comment:
Conceptually, the PR seems reasonable (once any doubts about temp table cross-contamination are put to rest). But in terms of code readability, I can't help but notice that if this method is called without `job_config`, we first create it containing only the destination at the start of the method, then we remove the destination here. Perhaps this could use a bit of a cleanup where the job_config management is all in one spot?

Reply:
strongly agree! but in this PR I'm trying to not break any other flows. Perhaps some users would prefer passing some of their own job configurations to `to_bigquery` method, that's why I only deal with the `destination` attribute.

Reply:
I believe we shouldn't allow users to pass the destination, there would be some changes needed but let's deal with them in a separate PR